doc/src/quickstart.md

```diff
@@ -71,49 +86,74 @@ calc.l:
 ```
 
-calc.y:
-```
+Roughly speaking, each line after the `%%` line is a regular expression (we use
+the [`regex`](https://crates.io/crates/regex) crate), a space character, and a
+quoted lexeme type name. In other words, if the user gives us input such as `234`
+we will create a single lexeme with a value (`234`) and a type (`INT`).
+
+The one exception is the final line: if a lexeme type name is replaced with ‘`;`’
+then any matching input is discarded. In this case, whitespace (tabs and spaces)
+is lexed, but no lexemes are created from it.
+
+
+## The grammar
+
+calc.y is as follows:
+```yacc
 %start Expr
-// Define the type that is to be returned by the actions. Can also be an enum.
-%type MYTYPE
+// Define the Rust type that is to be returned by the actions.
+%type u64
 %%
-Expr: Term 'PLUS' Expr { add($1, $3) }
+Expr: Term 'PLUS' Expr { $1 + $3 }
     | Term { $1 }
     ;
 
-Term: Factor 'MUL' Term { mul ($1, $3) }
+Term: Factor 'MUL' Term { $1 * $3 }
     | Factor { $1 }
     ;
 
 Factor: 'LBRACK' Expr 'RBRACK' { $2 }
-      | 'INT' { int($1) }
+      | 'INT' { parse_int($1) }
       ;
 %%
+// Any functions here are in scope for all the grammar actions above.
 
-type MYTYPE = u64;
-
-// The following functions are in scope for all the grammar actions above.
-
-fn int(s: &str) -> MYTYPE {
+fn parse_int(s: &str) -> u64 {
     match s.parse::<u64>() {
-    	Ok(val) => val as MYTYPE,
-	Err(_) => unreachable!()
+    	Ok(val) => val as u64,
+        Err(_) => panic!("{} cannot be represented as a u64", s)
     }
 }
+```
 
-fn add(arg1: u64, arg2: u64) -> u64 {
-    arg1 + arg2
-}
+The grammar is in 3 parts, separated by the `%%` lines.
 
-fn mul(arg1: u64, arg2: u64) -> u64 {
-    arg1 * arg2
-}
-```
+The first part specifies general settings for the grammar: its start rule
+(`Expr`) and the Rust type that actions in the grammar must produce (`u64`).
+
+The second part is the [Yacc
+grammar](http://dinosaur.compilertools.net/yacc/index.html). It consists of 3
+rules (`Expr`, `Term`, and `Factor`) and 6 productions (2 for each rule,
+separated by `|` characters). A production (sometimes called an “alternative”)
+consists of zero or more symbols. Symbols either reference rules or lexemes. If a
+production matches text, its ”action” (the Rust code between curly brackets at
+the end of the production) is executed. `lrpar`'s actions are slightly different
+to Yacc. The `$x` variables refer to the respective symbol in the production
+(i.e. `$1` refers to the first symbol in the production). If the symbol is a
+rule then an instance of the `%type` is stored in the `$x` variable; if the
+symbol is a lexeme then a string is stored instead.
 
-Now we can create the main function for our program which uses the parser
-generated in `build.rs`. When run, this program will take some user input,
-parse it according to the grammar, and return the result of the grammar
-actions:
+The third part is arbitrary Rust code which can be called by productions'
```

Comment:
maybe "by a productions action" sounds a bit better?

Reply:
I dunno. The current phrasing is, arguably, probably clearer?

Reply:
Sure, let's keep it as it is.